src/systems/filecoin_mining/storage_mining/storage_miner_actor.go

```diff
@@ -136,22 +151,66 @@ func (a *StorageMinerActorCode_I) _submitFaultReport(
 	rt.Abort("TODO") // TODO: Send(SPA, ProcessFaultReport(faultReport))
 	panic(faultReport)
 
+	if len(newDeclaredFaults) > 0 {
+		h, st := a.State(rt)
+		slashInfo := st._getDealSlashInfoFromSectors(rt, newDeclaredFaults.SectorsOn(), deal.SlashDeclaredFaults)
+		Release(rt, h, st)
+
+		// TODO: Send(StorageMarketActor, ProcessSectorDealSlash)
+		panic(slashInfo)
+	}
```

Comment:
there's a lot of code duplication here, which to me is usually a sign of error-prone-ness and need for decomposition, or similar